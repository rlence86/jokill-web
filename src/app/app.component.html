<div class="app__container">
    <app-header></app-header>
    <router-outlet></router-outlet>
    <app-footer></app-footer>
</div>