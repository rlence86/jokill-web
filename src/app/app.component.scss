@import 'src/styles.scss';
app {
    &__container {
        position: relative;
        min-height: 100vh;
    }
}