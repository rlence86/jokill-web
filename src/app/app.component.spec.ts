import { Component } from '@angular/core';
import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { Router } from '@angular/router';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';

@Component({
  template: '<div id="dummy-element">Dummy element</div>'
})
class DummyComponent {
}

@Component({
  selector: 'app-header',
  template: '<div id="header-element">Header element</div>'
})
class DummyHeaderComponent {
}

@Component({
  selector: 'app-footer',
  template: '<div id="footer-element">Footer element</div>'
})
class DummyFooterComponent {
}

describe('AppComponent', () => {
  let router: Router;
  let app: AppComponent;
  let fixture: ComponentFixture<AppComponent>;
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule.withRoutes([
          { path: '', component: DummyComponent }
        ])
      ],
      declarations: [
        AppComponent,
        DummyComponent,
        DummyHeaderComponent,
        DummyFooterComponent
      ],
    }).compileComponents();
    router = TestBed.inject(Router);
    fixture = TestBed.createComponent(AppComponent);
    app = fixture.componentInstance;
  });

  it('should create the app', () => {
    expect(app).toBeTruthy();
  });

  it(`should have a header and a footer'`, () => {
    const childHeadElement = fixture.debugElement.query(By.css('#header-element'));
    expect(childHeadElement.nativeElement.textContent).toBe('Header element');
    const childFooterlement = fixture.debugElement.query(By.css('#footer-element'));
    expect(childFooterlement.nativeElement.textContent).toBe('Footer element');
  });

  it(`should have routing working`, fakeAsync(() => {
    router.navigateByUrl('');
    tick();
    const childRouter = fixture.debugElement.query(By.css('#dummy-element'));
    expect(childRouter.nativeElement.textContent).toBe('Dummy element');
  }));

  it(`Renders correctly`, fakeAsync(() => {
    router.navigateByUrl('');
    tick();
    expect(fixture).toMatchSnapshot();
  }));

});
