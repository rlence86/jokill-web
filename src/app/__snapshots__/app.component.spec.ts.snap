// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent Renders correctly 1`] = `
<app-root>
  <div
    class="app__container"
  >
    <app-header>
      <div
        id="header-element"
      >
        Header element
      </div>
    </app-header>
    <router-outlet />
    <ng-component>
      <div
        id="dummy-element"
      >
        Dummy element
      </div>
    </ng-component>
    <app-footer>
      <div
        id="footer-element"
      >
        Footer element
      </div>
    </app-footer>
  </div>
</app-root>
`;
